Emby.Server.Implementations/SocketSharp/WebSocketSharpRequest.cs

```diff
@@ -417,51 +391,48 @@ private static string ResolvePathInfoFromMappedPath(string fullPath, string mapp
             return path.Length > 1 ? path.TrimEnd('/') : "/";
         }
 
-        private Dictionary<string, System.Net.Cookie> cookies;
-        public IDictionary<string, System.Net.Cookie> Cookies
-        {
-            get
-            {
-                if (cookies == null)
-                {
-                    cookies = new Dictionary<string, System.Net.Cookie>();
-                    foreach (var cookie in this.request.Cookies)
-                    {
-                        var httpCookie = (System.Net.Cookie)cookie;
-                        cookies[httpCookie.Name] = new System.Net.Cookie(httpCookie.Name, httpCookie.Value, httpCookie.Path, httpCookie.Domain);
-                    }
-                }
+        public string UserAgent => request.Headers[HeaderNames.UserAgent];
 
-                return cookies;
-            }
-        }
-
-        public string UserAgent => request.UserAgent;
+        public IHeaderDictionary Headers => request.Headers;
 
-        public QueryParamCollection Headers => request.Headers;
+        public IQueryCollection QueryString => request.Query;
 
-        private QueryParamCollection queryString;
-        public QueryParamCollection QueryString => queryString ?? (queryString = MyHttpUtility.ParseQueryString(request.Url.Query));
-
-        public bool IsLocal => request.IsLocal;
+        public bool IsLocal => string.Equals(request.HttpContext.Connection.LocalIpAddress.ToString(), request.HttpContext.Connection.RemoteIpAddress.ToString());
 
         private string httpMethod;
         public string HttpMethod =>
             httpMethod
-            ?? (httpMethod = request.HttpMethod);
+            ?? (httpMethod = request.Method);
 
         public string Verb => HttpMethod;
 
         public string ContentType => request.ContentType;
 
-        private Encoding contentEncoding;
-        public Encoding ContentEncoding
+        private Encoding ContentEncoding
         {
-            get => contentEncoding ?? request.ContentEncoding;
-            set => contentEncoding = value;
+            get
+            {
+                // TODO is this necessary?
+                if (UserAgent != null && CultureInfo.InvariantCulture.CompareInfo.IsPrefix(UserAgent, "UP"))
```

Comment:
what is this thing and where did it come from?

Reply:
It was taken elsewhere, so I didn't write it.